probe-rs/src/core/registers.rs

```diff
@@ -13,25 +17,86 @@ pub enum RegisterDataType {
     FloatingPoint,
 }
 
-/// Describes a register with its properties.
+/// This is used to label the register with a specific role that it plays during program execution and exception handling.
+/// The intention here is to harmonize the actual purpose of a register (e.g. `return address`),
```

Comment:
This reads a bit like a comment for the code reviewer while this imo should be userdocumentation and sound more like

```suggestion
/// This denotes the purpose of a register (e.g. `return address`),
```